### DISCUSSION TITLE
Problem using scintillua

### DISCUSSION THREAD

**Discussion by safe60** [orbitalquark/scintillua#31] [Q&A]
Hello everyone. This is my first post. I'm trying to use scintillua with scite 4.4.6. I think the istallation was correct but wasn't able to get any lexing. This not only for the lexer I'm trying to write but for all others lexers also. 

I have a 64 biut architecture but I'm using 32 bit version of LexLPeg because other one simply make scite to suddenly close without any error.  Any idea about what to check ?

**Comment by orbitalquark** [comment]
The architecture needs to match, so if you're using 64-bit SciTE, you need to use 64-bit LexLPeg.dll.

1. Downloaded SciTE 64-bit Windows binary.
2. Downloaded latest Scintillua release.
3. Moved scintillua's *lexers/* directory into SciTE directory.
4. Edited *SciTEGlobal.properties* to append `import lexers/lpeg` at the end.
5. Started SciTE.exe, and I have lexing for the default cpp lexer.

If you want to use 32-bit SciTE executable, then you'd also need to edit *lexers/lpeg.properties* and change `LexLPeg.dll` to `LexLPeg32.dll` where it appears.

Hope that helps.

**Reply by safe60** [reply]
Many thanks for your answer. It was a version problem actually. I confirm I was using 64 bit version of scite and the scenario was the one described in my previous post : with LexLPeg.dll. scite 64 bit  first open the main window and after.. let's say 10 seconds the main window is closed. There are no errors or warning messages. 

I switched to 32 bit version of Scite and everything worked fine since first run, including my very first lexer. I'm going to further enhance it.
thanks a lot.
